lib/cli/src/frameworks/web-components/Button.ts

```diff
@@ -1,10 +1,23 @@
 import { html } from 'lit-html';
-import './button.css';
+
+export interface ButtonProps {
+  primary: boolean;
+  backgroundColor: string;
+  size: string;
+  label: string;
+  onClick: () => void;
+}
 
 /**
  * Primary UI component for user interaction
  */
-export const Button = ({ primary, backgroundColor, size, label, onClick }) => {
+export const Button = ({
+  primary,
+  backgroundColor,
+  size,
+  label,
+  onClick,
+}: Partial<ButtonProps>) => {
```

Comment:
Best to use optional properties than Partial, you can reuse types from React framework

Reply:
The reason is that it is not the same.
![image](https://user-images.githubusercontent.com/599163/91608504-fa6b4180-e975-11ea-8bb2-ed8013f7daec.png)
I'm curious to see the reasons of having everything Partial. 
Not sure I understand the "both/duplicated" but we do want all types to be the same in every framework.

Reply:
i don't mind either way, here we can use an optional interface for consistency.

in reality a `Partial<T>` is probably just a `[K in keyof T]?: T[K]` anyway which i suppose is the same as a fully optional interface. 

i'm struggling to recall where and why i've seen them frowned upon now that i think about it. they are equivalent pretty much, and i can't find any such lint rules anymore against them. so i have no clue where i've picked this up from! maybe i should start using fully optional interfaces again :joy: